src/iterators.jl

```diff
@@ -113,7 +113,7 @@ cellid(cc::CellCache) = cc.cellid[]
 celldofs!(v::Vector, cc::CellCache) = copyto!(v, cc.dofs) # celldofs!(v, cc.dh, cc.cellid[])
 
 # TODO: These should really be replaced with something better...
-nfaces(cc::CellCache) = nfaces(eltype(cc.grid.cells))
+nfaces(cc::CellCache) = nfaces(getcelltype(cc.grid))
```

Comment:
This fails for non-concrete grids, but unrelated to this pr...
But should be relevant for merging of dofhandlers, cc: @kimauth

Reply:
But should be easy to fix.
```suggestion
nfaces(cc::CellCache) = nfaces(getcelltype(cc.grid, first(cc.cellid.x))
```

Reply:
Yes, I'm aware of this. Not an issue for merging, as this will still work as long as the grid is concrete (so it works for people who are using `DofHandler` now.
Should benchmark that solution, it has the usual problem of extracting an unconcrete cell, which might or might not be an issue.
I think `nfaces(fv::FaceValues)` should be used instead in most cases.